modules/standard/Curl.chpl

```diff
@@ -17,24 +17,326 @@
+  // disconnect and free the curl handle
+  c.close();
+
+
+Example 5
+---------
+
+.. code-block:: chapel
+
+  // This example uses the second interface + the first interface
+
+  var c = open(url="http://example.com", mode=iomode.r);
+  var str:string;
+  // Set verbose output from curl
+  c.setopt(curlopt_verbose, true);
+
+  // now read into the string
+  var reader = c.reader();
+  reader.readstring(str);
+  writeln(str);
+  reader.close();
+  c.close();
+
+
+Example 6
+---------
+
+.. code-block:: chapel
+
+  // Connect to an IMAP site, and fetch mail from the inbox
+  config const username = "user";
+  config const password = "xxxx";
+  config const imapSite = "your_imap_site_here";
+
+  var handle = open(url=imapSite+"/INBOX/;UID=1", mode=iomode.r);
+  var reader = handle.reader();
+  var str:string;
+
+  handle.setopt((curlopt_username, username),
+                (curlopt_password, password));
+
+  //Calling this would give the same output
+  // handle.perform();
+
+  reader.readstring(str);
+  write(str);
+
+  reader.close();
+  handle.close();
+
+.. _curl-example-7:
+
+Example 7
+---------
+
+.. code-block:: chapel
+
+  // This example shows a more complex example of how the two interfaces can work
+  // together: We use the second (setopt) interface to set the various options we
+  // need on a given curl handle, and then open a writer on that handle to write
+  // out via curl.
+
+  config const url      = "smtp goes here";
+
+  config const from     = "<some-email>";
+  config const to       = "<some-email>";
+  config const cc       = "<some-email>";
+
+  config const username = "username";
+  config const password = "password";
+
+  config const subject  = "Testing Curl in Chapel (SMTP)";
+  config const message  = "Hello! This is a message sent via Chapel!";
+
+
+  var handle = open(url=url, mode=iomode.cw);
+
+  var recipients:slist;
+  recipients.append(to);
+  recipients.append(cc);
+
+  var arr = "To: " + to + "\r\n" +
+            "From: " + from + "\r\n" +
+            "Cc: " + cc + "\r\n" +
+            "Subject: " + subject + "\r\n" + "\r\n" + message;
+
+
+  handle.setopt((curlopt_username       , username),
+                (curlopt_password       , password),
+                (curlopt_mail_from      , from),
+                (curlopt_use_ssl        , 3),
+                (curlopt_ssl_verifypeer , false),
+                (curlopt_ssl_verifyhost , false),
+                (curlopt_mail_rcpt      , recipients.list),
+                (curlopt_verbose        , true));
+
+  // Create a writer on the curl handle.
+  var writer = handle.writer();
+
+  // Now write out to Curl to send the email
+  writer.write(arr);
+
+  // recipients.free(); // BAD: This will free the data while it is in use!
+  writer.close();
+  // Note here how we free the slist AFTER we close the writer. This way we can
+  // ensure that we do not free data that is currently in use by curl.
+  recipients.free();
+  handle.close();
+
```

Comment:
Since this description is so long and involves many headers, it might be nice to add a header for the code definitions as well at the end here.